app/assets/javascripts/item_approval.js

```diff
@@ -0,0 +1,55 @@
+var itemApproval = (function() {
+  var defaultOptions = {
+    buttonSelector: '[data-behavior="item-approval"]',
+    approverInfoSelector: '[data-behavior="approver-information"]'
+  };
+
+  return {
+    init: function(opts) {
+      var _this = this;
+      _this.options = $.extend(defaultOptions, opts);
+      $(document).on('ready page:load', function(){
+        _this.addApprovalBehavior();
+      });
+    },
+
+    options: {},
+
+    addApprovalBehavior: function() {
+      var _this = this;
+      $(document).on('click', _this.options.buttonSelector, function() {
+        var row = $(this).closest('tr');
+        if(!_this.rowIsApproved(row)) {
+          _this.approveItem($(this));
+        }
+      });
+    },
+
+    approveItem: function(item) {
+      var _this = this;
+      var url = item.data('item-approval-url');
+      $.ajax(url).success(function(data) {
+         _this.markRowAsApproved(item);
+         _this.updateApproverInformation(item, data);
+       }).fail(function() {
+         alert('The item was not able to be approved.');
```

Comment:
We don't have proper failure handling designed yet.  There will most likely be a message put into the mediation table, so the `alert` is just a placeholder for the moment.